### PR TITLE
[Snapcraft] license

### PR DESCRIPTION
**Pull request by davide-pi**
ISSUE: #10299

Add Snapcraft license.
Create the `SnapcraftBase` from which other Snapcraft services can inherit.

Update Snapcraft version service to inherit from the `SnapcraftBase`.

Add Snapcraft base URL in configurations.

### CODE REVIEW

File: config/default.yml [hunk 1]
```diff
@@ -24,6 +24,8 @@ public:
       authorizedOrigins: 'https://api.opensuse.org'
     pypi:
       baseUri: 'https://pypi.org'
+    snapcraft:
+      baseUri: 'https://api.snapcraft.io/v2/snaps/info'
```
Comment (chris48s) [inline config/default.yml:28]: Does this actually need to be a setting?
AFAIK self-hosted snapcraft isn't a thing. Snapd is tied to https://snapcraft.io/
If this is something that needs to be configurable, can you link me to some docs about this
If it isn't, lets bin all this and hard-code the base url in `SnapcraftBase`

**Review by chris48s — COMMENTED** [review]
Thanks this looks pretty solid. Just one thing needing to follow up

File: config/default.yml (hunk 1 above)
Comment (davide-pi) [reply config/default.yml]: You're right it is better to maintain in the base service.
I was wondering if someone would make tests using a local endpoint but in this case is sufficient to change the address inside the service base class!

File: services/snapcraft/snapcraft-base.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import { BaseJsonService, pathParam } from '../index.js'
+
+export const snapcraftPackageParam = pathParam({
+  name: 'package',
+  example: 'Redis',
```
Comment (chris48s) [inline services/snapcraft/snapcraft-base.js:5]: One more quick thing I noticed. Package names are case sensitive Redis with a captial R is not found

```suggestion
  example: 'redis',
```

**Review by chris48s — APPROVED** [review]

**Pull request merged**